packages/mui-material/src/Modal/Modal.js

```diff
@@ -193,10 +198,12 @@ const Modal = React.forwardRef(function Modal(inProps, ref) {
         },
       });
     },
-    className: clsx(backdropSlotProps?.className, BackdropProps?.className, classes?.backdrop),
+    className: clsx(BackdropProps?.className, classes?.backdrop),
     ownerState,
   });
 
+  const backdropRef = useForkRef(BackdropProps?.ref, backdropProps.ref);
```

Comment:
@siriwatknp should `useSlot` merge the `ref` coming from `additionalProps`?

Reply:
@DiegoAndai  As jun is busy, i think we can move forward with existing approach and refactor `useSlot` later if we observe same pattern in other components